docs/docs/create-transformer-plugin.md

```diff
@@ -2,36 +2,272 @@
+  Gatsby calls [nodes](/docs/node-interface/).
+- **Transformer** plugins "transform" data provided by source plugins into new
+  nodes and/or node fields.
 
-The [`gatsby-transformer-remark`](/packages/gatsby-transformer-remark/) plugin
-"transforms" markdown nodes (fetched for example from the filesystem source plugin) into html and provides additional node details such as for example the `timeToRead` the `tableOfContents` and the `excerpt`.
+This doc is concerned with transformer plugins.
 
 ## What do transformer plugins do?
 
-Each transformer plugin is responsible for "enhancing" the existing source nodes they
-are provided to provide additional information. Therefore, you'll often use both source plugins and transformer plugins in your Gatsby sites.
+Transformer plugins "transform" data of one type into another type. You'll often use both source plugins and transformer plugins in your Gatsby sites.
+
+This loose coupling between the data source and transformer plugins allow Gatsby site builders to quickly assemble complex data transformation pipelines with little work.
+
+## Transformer plugin example
 
-This loose coupling between the data source and transformer plugins allow Gatsby
-site builders to quickly assemble complex data transformation pipelines with
-little work.
+The [`gatsby-transformer-remark`](/packages/gatsby-transformer-remark/) plugin "transforms" markdown nodes (fetched for example from the filesystem source plugin) into html and provides additional node details such as for example the `timeToRead` the `tableOfContents` and the `excerpt`.
 
 ## What does the code look like?
 
-Just like a source plugin, a transformer plugin is a normal NPM package. It has a `package.json` file with optional
-dependencies as well as a `gatsby-node.js` file where you implement Gatsby's Node.js
-APIs. Transformer plugins make use of the `setFieldsOnGraphQLNodeType` function and they commonly provide this functionality in the `extend-node-type` file.
+Just like a source plugin, a transformer plugin is a normal NPM package. It has a `package.json` file with optional dependencies as well as a `gatsby-node.js` file where you implement Gatsby's Node.js APIs.
+
+`gatsby-transformer-yaml` is transformer plugin that looks for new nodes with a media type of text/yaml (e.g. a .yaml file) and creates new YAML child node(s) by parsing the YAML source into JavaScript objects.
+
+As an example, let's partially rebuild a simplified `gatsby-transformer-yaml` directly in an example site. Let's say we have a default gatsby starter site which includes a YAML file:
+
+```yaml:title=src/data/example.yml
+- name: Jane Doe
+  bio: Developer based in Somewhere, USA
+- name: John Smith
+  bio: Developer based in Maintown, USA
+```
+
+### Make sure the data is sourced
 
-Your `gatsby-node.js` should look something like:
+First, we make sure the file is sourced, using `gatsby-source-filesystem` in `gatsby-config.js`:
```

Comment:
I'm thinking "you" could still be an advantageous point of view for these docs for consistency and clarity (to focus on the reader as the agent/actor who does the actions).

Reply:
👍

Reply:
Perhaps instead of saying "sourced" we make the process explicit here e.g. `gatsby-source-filesystem` creates `File` nodes which are exposed in our graphql schema which we can query. That'll help clarify that when we transform the File nodes into another node type, those too will automatically be exposed through GraphQL.